### DISCUSSION TITLE
password last change / integration with roundcube

### DISCUSSION THREAD

**Discussion by pedrossian** [postfixadmin/postfixadmin#692] [Q&A]
Hello,

There is a scrip for checking passwd for expiration which works fine.
In the situation of integration with roundcube the mailbox table has a field called passwordlastchange.
This is update by roundcube at password change from roundcube but the field is not update when changing the password from postfixadmin.

I am not good at smarty templates, can you help me with an advice how can i change the postfix admin to add also passwordlast change to be updated when the password is changed?

In my case is a mysql db.

Thank you

**Comment by DavidGoodwin** [comment]
I think you'd be better off doing it with a database trigger ? 

Gemini gives me something like this for MySQL which might serve as a starting point.

```
DELIMITER $$

CREATE TRIGGER update_password_timestamp
    BEFORE UPDATE ON mailbox
    FOR EACH ROW
BEGIN
    -- Check if password field has actually changed
    IF OLD.password != NEW.password THEN
        SET NEW.passwordlastchanged = NOW();
    END IF;
END$$

DELIMITER ;
```
